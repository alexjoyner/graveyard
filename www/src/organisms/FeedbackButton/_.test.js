import React from 'react';
import { shallow } from 'enzyme';
import { FeedbackButton } from '.';

describe('FeedbackButton Component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<FeedbackButton />);
  });
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should link to voicir feedback', () => {
    window.location.assign = jest.fn();
    const Button = component.find('Button');
    Button.props().onClick();
    expect(window.location.assign)
      .toBeCalledWith('http://feedback.voicir.com');
  });
});
