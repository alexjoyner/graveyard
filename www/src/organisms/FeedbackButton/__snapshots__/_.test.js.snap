// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackButton Component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeedbackButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Button
        color="primary"
        ghost={false}
        size="default"
      >
        Feedback
      </Button>,
      "style": Object {
        "position": "absolute",
        "right": "3px",
      },
      "to": "http://feedback.voicir.com",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Feedback",
        "color": "primary",
        "ghost": false,
        "size": "default",
      },
      "ref": null,
      "rendered": "Feedback",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Button
          color="primary"
          ghost={false}
          size="default"
        >
          Feedback
        </Button>,
        "style": Object {
          "position": "absolute",
          "right": "3px",
        },
        "to": "http://feedback.voicir.com",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Feedback",
          "color": "primary",
          "ghost": false,
          "size": "default",
        },
        "ref": null,
        "rendered": "Feedback",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
